std/net/curl.d

```diff
@@ -627,6 +627,61 @@ unittest
     assert(res == cast(ubyte[])[17, 27, 35, 41]);
 }
 
+/* Encode associative array using www-form-urlencoding
+ *  perhaps this should live in std.uri
+ */
+private string enc()(in string[string] values)
+{
+    string enc;
+    foreach (k, v; values)
+    {
+        if (enc.length)
+            enc ~= '&';
+        import std.uri;
+        enc ~= encodeComponent(k) ~ "=" ~ encodeComponent(v);
+    }
+    return enc;
```

Comment:
Not exactly, more something like:
```D
private string enc()(in string[string] values)
{
    Appender!string enc;
    enc.reseve(values.length*512);
    foreach (k, v; values)
    {
        // appending...
    }
    return enc.data;
}
```
to determine if the for the ampersand has to be added you can use a local size_t that you increments in the loop.